game-headed/src/main/java/org/triplea/game/client/ui/javafx/screens/MapSelection.java

```diff
@@ -0,0 +1,149 @@
+package org.triplea.game.client.ui.javafx.screens;
+
+import com.google.common.annotations.VisibleForTesting;
+import games.strategy.engine.framework.ui.GameChooserEntry;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContentDisplay;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.TilePane;
+import javafx.scene.web.WebView;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.triplea.game.client.parser.DefaultGameDetector;
+import org.triplea.game.client.parser.GameDetector;
+import org.triplea.game.client.ui.javafx.screen.ControlledScreen;
+import org.triplea.game.client.ui.javafx.screen.ScreenController;
+import org.triplea.game.client.ui.javafx.util.FxmlManager;
+import org.triplea.util.LocalizeHtml;
+
+public class MapSelection implements ControlledScreen<ScreenController<FxmlManager>> {
```

Comment:
High-level / important: I strongly suspect this class should have almost no logic in it. Similar to how http controllers delegate most everything, a similar pattern here could help us avoid having to include FX components in test. More specifically, if all logic were delegated, then we'd just have to ensure the delegated logic is invoked and we could test that logic independently and not be coupled to UI layer. WDYT?

Reply:
Well in an ideal world this class should define what all the UI components inside this class are actually doing, so all of the UI logic regarding this exact screen should be here